Source/ProjectRimFactory/SAL3/Things/Assemblers/Building_ProgrammableAssembler.cs

```diff
@@ -539,6 +543,10 @@ public List<Thing> GetThingQueue()
         // thingQueue is List to save properly
         //   List of produced things, waiting to be placed:
         protected List<Thing> thingQueue = new List<Thing>();
+        //max number of items that can be stored in thingQueue before production is halted
+        protected const int max_thingQueue_Count = 100;
+
+        protected bool allowProduction_thingQueue => thingQueue.Count < max_thingQueue_Count;

```

Comment:
C# Convention would be AllowProduction_thingQueue => etc (capital letter; reserve lowercase for actual member fields)